File: implementations/c/summation/summation_mid.c
#include <stdio.h>
#include <stdint.h>
#include <inttypes.h>

#define N 10000

static int64_t arr[N];

static void setup(void) {
    for (int i = 0; i < N; i++) {
        arr[i] = (int64_t)i + 1;
    }
}

static int64_t summation(void) {
    int64_t sum = 0;
    for (int i = 0; i < N; i++) {
        sum += arr[i];
    }
    return sum;
}

int main(void) {
    setup();

    printf("ready\n");
    fflush(stdout);

    char buf[64];
    while (fgets(buf, sizeof(buf), stdin)) {
        printf("%" PRId64 "\n", summation());
        fflush(stdout);
    }

    return 0;
}

File: implementations/python/summation/summation_mid.py
import sys

N = 10000

arr = [0] * N

def setup():
    for i in range(N):
        arr[i] = i + 1

def summation():
    total = 0
    for i in range(N):
        total += arr[i]
    return total

def main():
    setup()

    sys.stdout.write("ready\n")
    sys.stdout.flush()

    for line in sys.stdin:
        sys.stdout.write(str(summation()) + "\n")
        sys.stdout.flush()

if __name__ == "__main__":
    main()